MSAL/src/public/MSALWebviewParameters.h

```diff
@@ -0,0 +1,71 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import <Foundation/Foundation.h>
+#import <WebKit/WebKit.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MSALWebviewParameters : NSObject <NSCopying>
+
+#if TARGET_OS_IPHONE
+/*!
+ The view controller to present from. If nil, the current topmost view controller will be used.
+ */
+@property (nullable, weak, nonatomic) UIViewController *parentViewController;
+
+/*!
+ Modal presentation style for displaying authentication web content.
+ */
+@property (nonatomic) UIModalPresentationStyle presentationStyle;
+
+#endif
+
+/*!
+ A specific webView type for the interactive authentication flow.
+ By default, it will be set to MSALGlobalConfig.defaultWebviewType.
+ */
+@property (nonatomic) MSALWebviewType webviewType;
+
+/*!
+ For a webviewType MSALWebviewTypeWKWebView, custom WKWebView can be passed on.
+ Web content will be rendered onto this view.
+ Observe strings declared in MSALPublicClientStatusNotifications to know when to dismiss.
+ */
+@property (nonatomic, nullable) WKWebView *customWebview;
+
+#if TARGET_OS_IPHONE
+- (nonnull instancetype)initWithParentViewController:(UIViewController *)parentViewController;
```

Comment:
it feels weird for macOS - you need to create new instance and pass it... Seems unnecessary API for macOS if we introduce it. Will macOS also require parent controller? If not, do we want to make this optional for macOS?

Reply:
done